Mathlib/CategoryTheory/Comma/Final.lean

```diff
@@ -132,6 +136,30 @@ lemma map_final {A : Type u₁} [Category.{v₁} A] {B : Type u₂} [Category.{u
   rw [IsIso.Iso.inv_inv]
   infer_instance⟩
 
+section Filtered
+
+variable {A : Type u₁} [Category.{v₁} A]
+variable {B : Type u₂} [Category.{u₂} B]
+variable {T : Type u₂} [Category.{u₂} T]
+variable (L : A ⥤ T) (R : B ⥤ T)
+
+attribute [local instance] map_final in
+/-- Let `A` and `B` be filtered categories, `R : B ⥤ T` be final and `L : A ⥤ T`. Then, the
+comma category `Comma L R` is filtered. -/
+lemma filtered [IsFiltered A] [IsFiltered B] [R.Final] : IsFiltered (Comma L R) := by
+  haveI (a : A) : IsFiltered (Comma (fromPUnit (L.obj a)) R) :=
```

Comment:
Are these `haveI` really needed (recall that `haveI` in Lean 4 is not the same as `haveI` in Lean 3...)?

Reply:
No, they can as well be `have`s I think.